docs/zephyr_02_functional_requirements.sdoc

```diff
@@ -448,6 +448,448 @@ Zephyr shall provide an interface to assign a specific handler for a fatal error
 
 [/SECTION]
 
+[SECTION]
+TITLE: System Initialization
+
+[REQUIREMENT]
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: Initialization follows dependency order
+STATEMENT: >>>
+Zephyr shall initialize the system following the order of component dependencies.
+<<<
+
+[REQUIREMENT]
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: Data provided from preceding executable available throughout initialization process
+STATEMENT: >>>
+Zephyr shall expose the data provided by the preceding executable throughout the initialization of the Zephyr OS.
+<<<
+
+[REQUIREMENT]
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: Inherited CPU stack usage kept within published limits
+STATEMENT: >>>
+Zephyr shall use no more than the published amount of the CPU stack inherited from the prior executable.
+<<<
+
+[REQUIREMENT]
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: Sustain hardware watchdog inherited from preceding executable
+STATEMENT: >>>
+Zephyr shall sustain the active hardware watchdog inherited from the preceding executable.
+<<<
+
+[SECTION]
+TITLE: Initialization on Inherited CPU Stack
+
+[REQUIREMENT]
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: Pre-CPU Architecture Initialization notification
+STATEMENT: >>>
+Zephyr shall publish an notification to each registered component
+after the C domain has been set up and
+before any CPU-architecture specific initialization.
+<<<
+
+[SECTION]
+TITLE: CPU Architecture Specific Initialization
+
+[REQUIREMENT]
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: CPU-architecture initialization invalidates CPU caches
+STATEMENT: >>>
+Zephyr shall invalidate the CPU caches during system initialization.
+<<<
+
+[REQUIREMENT]
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: CPU-architecture initialization disables memory protection config inherited from preceding executable
+STATEMENT: >>>
+Zephyr shall disable the CPU memory protection mechanisms inherited from the preceding executable.
```

Comment:
The first-stage bootloader may not use the memory protection, or it may change the memory protection config while flashing. I think the loaded image should not require the loader to set up memory for the runtime of the loaded image.